test/suites/dev/moonbase/test-xcm/test-mock-hrmp-transact-ethereum-3.ts

```diff
@@ -0,0 +1,263 @@
+      contractDeployed = contractAddress;
+      contractABI = abi;
+
+      const { originAddress, descendOriginAddress } = descendOriginFromAddress20(context);
+      sendingAddress = originAddress;
+      descendedAddress = descendOriginAddress;
+      random = generateKeyringPair();
+
+      // registerOldForeignAsset
+      const { registeredAssetId, registeredAsset } = await registerOldForeignAsset(
+        context,
+        STATEMINT_LOCATION,
+        assetMetadata,
+        1_000_000_000_000
+      );
+      assetId = registeredAssetId;
+      expect(registeredAsset.owner.toHex()).to.eq(palletId.toLowerCase());
+    });
+
+    for (const xcmVersion of XCM_VERSIONS) {
+      it({
+        id: `T01-XCM-v${xcmVersion}`,
+        title: `should receive transact and should be able to execute (XCM v${xcmVersion})`,
+        test: async function () {
+          const config = {
+            assets: [
+              {
+                multilocation: ASSET_MULTILOCATION,
+                fungible: assetsToTransfer,
+              },
+            ],
+            beneficiary: descendedAddress,
+          };
+
+          // How much will the message weight?
+          const chargedWeight = await weightMessage(
+            context,
+            context
+              .polkadotJs()
+              .createType(
+                "XcmVersionedXcm",
+                new XcmFragment(config)
+                  .reserve_asset_deposited()
+                  .clear_origin()
+                  .buy_execution()
+                  .deposit_asset()
+                  .as_v3()
+              ) as any
+          );
+
+          // we modify the config now:
+          // we send assetsToTransfer plus whatever we will be charged in weight
+          config.assets[0].fungible = assetsToTransfer + chargedWeight;
+
+          // Construct the real message
+          const xcmMessage = new XcmFragment(config)
+            .reserve_asset_deposited()
+            .clear_origin()
+            .buy_execution()
+            .deposit_asset()
+            .as_v3();
+
+          // Send an XCM and create block to execute it
+          await injectHrmpMessageAndSeal(context, statemint_para_id, {
+            type: "XcmVersionedXcm",
+            payload: xcmMessage,
+          } as RawXcmMessage);
+
+          // Make sure descended address has the transferred foreign assets (minus the xcm fees).
+          expect(
+            (await context.polkadotJs().query.assets.account(assetId, descendedAddress))
+              .unwrap()
+              .balance.toBigInt()
+          ).to.eq(assetsToTransfer);
+
+          // Get initial contract count
+          const initialCount = (
+            await context.viem().call({
+              to: contractDeployed,
+              data: encodeFunctionData({ abi: contractABI, functionName: "count" }),
+            })
+          ).data;
+          const initialCountBigInt = BigInt(initialCount!.toString());
+          const GAS_LIMIT = 100_000;
```

Comment:
Here it seems to be missing a new test definition:

```typescript
it({
        id: `T01-XCM-v${xcmVersion}`,
        ...
```

The original test had it, but was missing the previous one (a whole test was happening within the `beforeAll`).

Reply:
No, this is expected. We are running the test 3 times, 1 time for each version. This prepares the state for each run